metrics/ci_integrations/test/client/buildkite/test_utils/mock/buildkite_mock_server.dart

```diff
@@ -126,6 +144,15 @@ class BuildkiteMockServer extends ApiMockServer {
     await MockServerUtils.writeResponse(request, body: response);
   }
 
+  /// Responses with a [BuildkiteBuild].
```

Comment:
Should we also set the correct build number as requested (provided in `request`)?